.github/workflows/test.yml

```diff
@@ -0,0 +1,70 @@
+name: test
+
+on:
+  pull_request:
+  push:
+    branches:
+      - master
+  workflow_dispatch:
+
+env:
+  GOOGLETEST_VERSION: "1.15.2"
+
+jobs:
+  ctest:
+    strategy:
+      matrix:
+        runs-on:
+          - ubuntu-latest
+          - macos-13
+          - macos-latest
+      fail-fast: false
+    runs-on: ${{ matrix.runs-on }}
+    steps:
+      - uses: zacharyburnett/setup-abseil-cpp@de39f445295c887839e30c864ffbbb1c0231bc83 # 1.0.5
+        with:
+          cmake-build-args: "-DCMAKE_CXX_STANDARD=17 -DABSL_PROPAGATE_CXX_STD=ON -DABSL_ENABLE_INSTALL=ON -DBUILD_TESTING=off -DCMAKE_POSITION_INDEPENDENT_CODE=ON"
+          abseil-version: "20240722.0"
+      - name: retrieve googletest v${{ env.GOOGLETEST_VERSION }}
+        run: |
+          wget https://github.com/google/googletest/releases/download/v${{ env.GOOGLETEST_VERSION }}/googletest-${{ env.GOOGLETEST_VERSION }}.tar.gz
+          tar -xzvf googletest-${{ env.GOOGLETEST_VERSION }}.tar.gz
+          echo GOOGLETEST_ROOT=${{ runner.temp }}/googletest-${{ env.GOOGLETEST_VERSION }} >> $GITHUB_ENV
+        working-directory: ${{ runner.temp }}
+      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+      - run: mkdir build
+      - run: cmake -DCMAKE_CXX_STANDARD=17 -DCMAKE_PREFIX_PATH=/usr/local/ -DGOOGLETEST_ROOT=${{ env.GOOGLETEST_ROOT }} -DGOOGLETEST_VERSION=${{ env.GOOGLETEST_VERSION }} ..
+        working-directory: build/
+      - run: cmake --build . --parallel
+        working-directory: build/
+      - if: always()
+        run: cmake --build . --parallel --target=test
```

Comment:
Can you set `CTEST_OUTPUT_ON_FAILURE` or use `ctest --output-on-failure`? (I've never actually used ctest directly.)  It's helpful to see why tests fail.

https://cmake.org/cmake/help/latest/envvar/CTEST_OUTPUT_ON_FAILURE.html#envvar:CTEST_OUTPUT_ON_FAILURE

Reply:
I added the env var to https://github.com/zacharyburnett/s2geometry/actions/runs/11841933989/job/32999371748, and it seems to be producing more output. Is that helpful?

Reply:
Yes, thanks.